### PR TITLE
Ts migration

### PR DESCRIPTION
**Pull request by Hyperkid123**
jira: https://issues.redhat.com/browse/RHCLOUD-20636

### Changes

- Expose complete redux store state type
- Migrate Active component to TS
- Migrate Banner component to TS
- Migrate CrossRequest notifier and its dependencies to TS

### CODE REVIEW

File: src/js/App/CrossRequestNotifier/CrossRequestNotifier.tsx [hunk 1]
```diff
@@ -1,52 +1,47 @@
...
 );
 
-DescriptionComponent.propTypes = {
-  id: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.string), PropTypes.string]).isRequired,
-  markRead: PropTypes.func.isRequired,
-};
-
 const CrossRequestNotifier = () => {
-  const [, forceRender] = useReducer((state = false) => !state);
+  const [, forceRender] = useReducer((state) => state + 1, 0);
```
Comment (karelhala) [inline src/js/App/CrossRequestNotifier/CrossRequestNotifier.tsx:40]: Any reason for the `+1`?

**Review by karelhala — APPROVED** [review]
Looking good just a confirmation question.

File: src/js/App/CrossRequestNotifier/CrossRequestNotifier.tsx (hunk 1 above)
Comment (Hyperkid123) [reply src/js/App/CrossRequestNotifier/CrossRequestNotifier.tsx:40]: Yes. Actually using the `true/false` can sometimes run through multiple updates and will not resolve in a forced re-render. We had this issue in DDF and figured we can't use a simple boolean flip. Hence the counter.

Comment (karelhala) [reply src/js/App/CrossRequestNotifier/CrossRequestNotifier.tsx:40]: Okay, works for me

**Pull request merged**